tests/test_cli.py

```diff
@@ -119,3 +119,48 @@ def test_cli_wrong_definitions_name():
         ],
     )
     assert result_valid.exit_code == 1
+
+
+def test_cli_custom_dimensions():
+    """Check that CLI runs through with a non-default dimension"""
+
+    result_valid = runner.invoke(
+        cli,
+        [
+            "validate-project",
+            str(TEST_DATA_DIR / "non-default_dimensions_passing"),
+            "--dimensions",
+            "['variable', 'region', 'foo']",
+        ],
+    )
+    assert result_valid.exit_code == 0
+
+
+def test_cli_custom_dimensions_empty():
+    """Check that CLI raises an expected error with an empty non-default dimension"""
+
+    result_valid = runner.invoke(
+        cli,
+        [
+            "validate-project",
+            str(TEST_DATA_DIR / "non-default_dimensions_failing"),
+            "--dimensions",
+            "['variable', 'region', 'foo']",
+        ],
+    )
+    assert result_valid.exit_code == 1
+
+
+def test_cli_custom_dimensions_fails():
+    """Check that CLI raises expected error when given a non-existing dimension"""
```

Comment:
What is the difference between this test and the previous one? The way I read it is that in both cases you specify a non-existing dimension. 
This is speculation now, but could it be that for the previous example you have created a folder called `foo/` and it's empty? In this case this would explain the difference and since git only tracks files and not empty folders the folder does not show up here on GitHub. If this is indeed the case and you want to test for an empty dimensions one way around this is adding an empty file called `.gitkeep` to the folder `foo/`. See here for details https://stackoverflow.com/questions/7229885/what-are-the-differences-between-gitignore-and-gitkeep.

Reply:
Thanks! I think I actually messed up the names so maybe that's my fault there was no folder, but yes I can do that anyways. And yes, your assumptions were correct, I changed it so that it is more comprehensible.